### PR TITLE
level up markdowns and readme

### LINKED ISSUE #95: Level up README and related markdown content
Level up and cleanup markdown content. 

See [.github#8](https://github.com/opensearch-project/.github/issues/8) for details and recommendations, and [common-utils](https://github.com/opensearch-project/common-utils/) for an example.

- [x] README.md
- [x] CONTRIBUTING.md
- [x] DEVELOPER_GUIDE.md
- [x] MAINTAINERS.md
- [x] ADMINS.md
- [x] SECURITY.md
- [x] RELEASING.md
- [ ] TESTING.md
- [x] other .md files

### PR DESCRIPTION
**Pull request by zhongnansu**
Signed-off-by: Zhongnan Su <szhongna@amazon.com>

### Description
- level up markdowns and readme
- Add link checker github workflow

### Issues Resolved
#95 #96 

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: ADMINS.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+## Admins
+| Admin             | GitHub ID                                         | Affiliation |
+|------------------------|---------------------------------------------------|-------------|
+| Anantha Krishna Bhatta | [akbhatta](https://github.com/akbhatta)           | Amazon      |
+| David Cui              | [davidcui-amzn](https://github.com/davidcui-amzn) | Amazon      |
+| Joshua Li              | [joshuali925](https://github.com/joshuali925)     | Amazon      |
+| Zhongnan Su            | [zhongnansu](https://github.com/zhongnansu)       | Amazon      |
```
Comment (joshuali925) [inline ADMINS.md:7]: I believe admins are different from maintainers, currently we only have one admin https://github.com/opensearch-project/trace-analytics/blob/2aec3399bea4f07c2add94912a4b36da930efab8/ADMINS.md

File: RELEASE.md [hunk 2]
```diff
@@ -0,0 +1 @@
+This project follows [OpenSearch branching, labelling, and releasing](https://github.com/opensearch-project/.github/blob/main/RELEASING.md).
```
Comment (joshuali925) [inline RELEASE.md:1]: file name should be RELEASING.md?

Comment (zhongnansu) [reply RELEASE.md:1]: good catch

File: ADMINS.md (hunk 1 above)
Comment (zhongnansu) [reply ADMINS.md:7]: can we apply for admin access only for certain repo?

**Review by joshuali925 — APPROVED** [review]

**Comment by davidcui-amzn** [comment]
`DEVELOPER_GUIDE.md` is missing? Or is the separation of that document optional? I see it in your PRs and am planning to add it to mine @joshuali925

File: ADMINS.md [hunk 3]
```diff
@@ -0,0 +1,7 @@
+## Admins
+
+| Admin           | GitHub ID                               | Affiliation |
+| --------------- | --------------------------------------- | ----------- |
+| Henri Yandell   | [hyandell](https://github.com/hyandell) | Amazon      |
```
Comment (dblock) [inline ADMINS.md:5]: @CEHENKLE is this correct? Who should be listed as admin here?

**Review by dblock — APPROVED** [review]
I'd check-in with @CEHENKLE on admins and this is good to go.

**Comment by zhongnansu** [comment]
> `DEVELOPER_GUIDE.md` is missing? Or is the separation of that document optional? I see it in your PRs and am planning to add it to mine @joshuali925

fixed

**Review by davidcui-amzn — APPROVED** [review]

**Pull request merged**